packages/roosterjs-content-model-plugins/lib/tableEdit/editors/features/TableMover.ts

```diff
@@ -129,3 +157,212 @@ function isTableTopVisible(editor: IEditor, rect: Rect | null, contentDiv?: Node
 }
+
+function setTableMoverCursor(editor: IEditor, state: boolean, type?: 'move' | 'copy') {
+    editor?.setEditorStyle(TABLE_MOVER_STYLE_KEY, state ? 'cursor: ' + type ?? 'move' : null);
+}
+
+function onDragStart(context: TableMoverContext, event: MouseEvent) {
+    context.onStart();
+
+    const { editor, table, div } = context;
+
+    setTableMoverCursor(editor, true, 'move');
+
+    // Create table outline rectangle
+    const trect = table.getBoundingClientRect();
+    const createElementData = {
+        tag: 'div',
+        style: 'position: fixed; user-select: none; border: 1px solid #808080',
+    };
+    const tableRect = createElement(createElementData, document) as HTMLDivElement;
+    tableRect.style.width = `${trect.width}px`;
+    tableRect.style.height = `${trect.height}px`;
+    tableRect.style.top = `${trect.top}px`;
+    tableRect.style.left = `${trect.left}px`;
+    div.parentNode?.appendChild(tableRect);
+
+    // Get current selection
+    const initialSelection = editor.getDOMSelection();
+
+    // Select first cell of the table
+    editor.setDOMSelection({
+        type: 'table',
+        firstColumn: 0,
+        firstRow: 0,
+        lastColumn: 0,
+        lastRow: 0,
+        table: table,
+    });
+
+    // Get the table content model
+    const [cmTable] = getFirstSelectedTable(editor.getContentModelCopy('disconnected'));
+
+    // Restore selection
+    editor.setDOMSelection(initialSelection);
+
+    return {
+        cmTable,
+        initialSelection,
+        tableRect,
+    };
+}
+
+// Get range from coordinate.
+function getNodePositionFromEvent(editor: IEditor, x: number, y: number): DOMInsertPoint | null {
+    const doc = editor.getDocument();
+    const domHelper = editor.getDOMHelper();
+
+    if (doc.caretRangeFromPoint) {
+        // Chrome, Edge, Safari, Opera
+        const range = doc.caretRangeFromPoint(x, y);
+        if (range && domHelper.isNodeInEditor(range.startContainer)) {
+            return { node: range.startContainer, offset: range.startOffset };
+        }
+    }
+
+    if ('caretPositionFromPoint' in doc) {
+        // Firefox
+        const pos = (doc as any).caretPositionFromPoint(x, y);
+        if (pos && domHelper.isNodeInEditor(pos.offsetNode)) {
+            return { node: pos.offsetNode, offset: pos.offset };
+        }
+    }
+
+    if (doc.elementFromPoint) {
+        // Fallback
+        const element = doc.elementFromPoint(x, y);
+        if (element && domHelper.isNodeInEditor(element)) {
+            return { node: element, offset: 0 };
+        }
+    }
+
+    return null;
+}
+
+function onDragging(context: TableMoverContext, event: MouseEvent, initValue: TableMoverInitValue) {
+    const { tableRect } = initValue;
+    const { editor } = context;
+
+    // Move table outline rectangle
+    tableRect.style.top = `${event.clientY + TABLE_MOVER_LENGTH}px`;
+    tableRect.style.left = `${event.clientX + TABLE_MOVER_LENGTH}px`;
+
+    event.ctrlKey || event.metaKey
```

Comment:
Actually, Mac also has Ctrl key. So do we want to allow copy on Mac using Ctrl key or Meta key or both?
If we only want to do copy with Meta key here, we should also check if it is running on Mac then Ctrl key does not count.

Reply:
I just tried on Mac and it seems Mac never allows copy, even with Ctrl or Meta key. So I think we can keep it consistent here, only do copy when !isMac and ctrlKey

Reply:
So this means not allowing the feature on Mac?

Reply:
I mean on Mac we only allow move, but not copy. At least that is how normal drag and drop work on Mac. You can also confirm with @jvillalobos